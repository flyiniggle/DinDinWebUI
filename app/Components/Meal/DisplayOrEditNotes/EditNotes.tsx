import * as React from 'react';

interface IEditNotesProps {
    value?: string,
    onChange: (e: React.FormEvent) => void
}

class EditNotes extends React.Component<IEditNotesProps, {}> {
    private input = React.createRef<HTMLTextAreaElement>();

    componentDidmount = () => {
        //this.input.focus()
    }

    render() {
        const { value, onChange } = this.props;

        return <textarea ref={this.input} className="form-control d-block mb-2" onChange={onChange} value={value} />
    }
}

export { IEditNotesProps };
export default EditNotes;