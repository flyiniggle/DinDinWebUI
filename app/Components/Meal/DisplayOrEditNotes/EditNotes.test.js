import * as React from 'react';
import { shallow } from 'enzyme';

import EditNotes from './EditNotes';

describe('#Components #Meal #DisplayOrEditNotes #DisplayNotes', function() {
    it('should display the value if one is provided', function() {
        const onChangeSpy = jest.fn();
        const wrapper = shallow(<EditNotes value="wheeeee" onChange={ onChangeSpy } />);

        //expect(wrapper.text()).toEqual('wheeeee');
    });

    it('should call the changehandler when updating the value.', function() {
        const onChangeSpy = jest.fn();
        const wrapper = shallow(<EditNotes onChange={ onChangeSpy } />);

        wrapper.simulate('change');
        expect(onChangeSpy).toHaveBeenCalled();
    });
});