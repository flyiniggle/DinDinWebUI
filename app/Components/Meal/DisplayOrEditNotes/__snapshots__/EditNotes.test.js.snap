// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Components #Meal #DisplayOrEditNotes #DisplayNotes should display the value if one is provided 1`] = `"<textarea class=\\"form-control d-block mb-2\\">wheeeee</textarea>"`;
